### PR TITLE
Feature/by image utils

### PR DESCRIPTION
**Pull request by TobiasAdler**
# Description

**By-image-utils & helper methods**

In `DesktopWebDriverUtils`, the methods `clickByImage` and `mouseOverByImage` were added to provide a new way to hover over and click an element, that may not be possible to find in the usual ways (for example the test method elements in the threads view of testerra report, which are inside of a canvas element and can not be accessed by selenium).

These methods are build with the sikuli-api, which provides the possibility to search for images in the current viewport of the web driver. Since only the current viewport is analysed, a scroll-by-value-functionality and helper methods for the browser were added in `JSUtils` and used to ensure the image is searched in the complete webpage.

**Unit- & report-ng-tests**

The report-ng-tests for the threads view were updated to use the `mouseOverByImage`-method since due to the canvas element of echarts it was not possible to access the specific test methods in the usual ways.
Unit tests for the new utilities were added as well.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: driver-ui-desktop/build.gradle [hunk 1]
```diff
@@ -1,5 +1,8 @@
 dependencies {
     api driver_ui
+
+    // <ByImage, SikuliWebDriver>
+    implementation 'org.sikuli:sikuli-api:1.2.0'
```
Comment (martingrossmann) [inline driver-ui-desktop/build.gradle:5]: Sikuli is already part of `driver-ui/build.gradle`, Why it's needed here?

Comment (TobiasAdler) [reply driver-ui-desktop/build.gradle]: You are right, this implementation is redundant and not needed. It has been removed in the most recent commit.

**Comment by mreiche** [comment]
I would rather like to implement an `By.image('path/to/referenceImage.png')` than introducing utils for that. 

```java
UiElement uiElement = find(By.image('...'));
uiElement.click();
```

**Comment by martingrossmann** [comment]
A kind of `By.image` is still part of Testerra. It was taken over from the former framework but was never realy updated and documeneted. Maybe it can be removed.
Sikuli was used to find the given image in the viewport. With the help of the coordinates at this point the current existing Selenium Webelement was tried to identify. This was used to create a `GuiElement`.

But this usecase does not work at canvas graphics where you only have a big `canvas` Webelement. This solution works without any webelement and should be a little helper (e.g. for canvas elements), not more. 

From my point a `UiElement` always based on a Selenium Webelement to support all features like actions, asserts, ...

**Comment by martingrossmann** [comment]
@TobiasAdler I would approve this PR. Please resolve the conflicts. Thanks!

**Comment by TobiasAdler** [comment]
@martingrossmann All conflicts have been resolved.

**Review by mbeuthan — APPROVED** [review]

**Review by martingrossmann — APPROVED** [review]

**Pull request merged**